### PR TITLE
model(whisper): allocate extra block for prefill padding (vllm)

### PR DESCRIPTION
**Pull request by rebel-eunji**
# Pull Request Description

> **⚠️ Important: Branch Target**
> - **New features, enhancements, and non-critical fixes**: Merge to `dev` branch
> - **Critical hotfixes only**: Merge to `main` branch (must also merge to `dev`)
> 
> Please ensure you've selected the correct base branch before submitting!

## Type of Change

- [ ] Release (dev → main merge for production release)
- [ ] New Model Support
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring
- [x] Other (please describe):

## Changes Overview
whisper 모델에 한하여 블록 1개 추가 할당하는 PR입니다.

## Motivation and Context
For the Whisper model in vLLM, a dummy block is required as a scratch pad to run the encoder and the 1st decoder step together during the prefill step. To support this, the block allocation logic in optimum-rbln has been changed from `batch_size` to `batch_size + 1`.

This change updates `rbln_config.json` as shown below.
```
ex)
...
        [
          "self_key_value_states_11",
          [
            5,
            8,
            448,
            64
          ],
          "float32"
        ]
      ],
      "npu": null,
      "tensor_parallel_size": null
    }
  ],
  "batch_size": 4,
  "token_timestamps": false,
  "enc_max_seq_len": 1500,
  "dec_max_seq_len": 448,
  "use_attention_mask": false,
  "kvcache_num_blocks": 5,
  "kvcache_block_size": 448
}
```

## Related Issues

----
# Conventional commit
```
type(optional scope): description
```
----
# Type candidate
  - Model Updates
    - `model`: Adding New models or Bugfix for existing models
      - ex) Add LlavaNext 
      - ex) Bugfix Whisper
  - Enhancements
    - `performance`: Optimizing some models or this library itself
      - ex) Loading RBLNModel faster
      - ex) Optimizing Memory Usage of DecoderOnlyModel
  - Code Refactor
    - `refactor`: Re-arrange class architecture, or more.
      - ex) Refactor Seq2Seq
  - Documentation
    - `doc`: Update docstring only
  - Library Dependencies
    - `dependency`: Update requirements, something like that.
  - Release
    - `release`: Merging dev to main for production release
      - ex) Release v1.2.0
  - Other
    - `other`: None of above.
      - ex) ci update
      - ex) pdm update

### CODE REVIEW

File: src/optimum/rbln/transformers/models/whisper/modeling_whisper.py [hunk 1]
```diff
@@ -298,7 +298,7 @@ def _update_rbln_config(
                 "cross_key_value_states",
                 [
                     model_config.decoder_layers * 2,
-                    rbln_config.batch_size,
+                    rbln_config.batch_size + 1,
```
Comment (rebel-thkim) [inline src/optimum/rbln/transformers/models/whisper/modeling_whisper.py]: The `rbln_config` contains the following attribute:
```
            kvcache_num_blocks (int, optional): The total number of blocks to allocate for the
                PagedAttention KV cache for SelfAttention. Defaults to batch_size.
```
Instead of hard-coding it as `batch_size + 1`, why not use this attribute?
If you'd like, I can add commits to this PR.

**Review by rebel-thkim — DISMISSED** [review]

File: src/optimum/rbln/transformers/models/whisper/modeling_whisper.py (hunk 1 above)
Comment (rebel-eunji) [reply src/optimum/rbln/transformers/models/whisper/modeling_whisper.py]: That would be great, thanks! Please go ahead.

Comment (rebel-thkim) [reply src/optimum/rbln/transformers/models/whisper/modeling_whisper.py]: @rebel-eunji, Done

**Comment by rebel-eunji** [comment]
It raises following compile error. @rebel-thkim 
```
[2026-05-14 16:27:12] [2160342][2160342] [I] [RSDSharding] Running on single node.
loc(callsite("-":33:3 at fused<array<i64: 340>>["rbln_cache_update":0:0])): error: 'rtosa.memory.store' op [verify] Axes not specified in the accessor must be the same.
[2026-05-14 16:27:12] [2160342][2160342] [E] Failed on shape inference on "rtosa.memory.store"(%320, %321, %322) <{accessors = #common.memory_accessors<[#common.memory_accessor<1, <true, 0>, <false, 1>, 1>]>}> : (tensor<12x5x8x448x64xdlf16>, tensor<12x1x8x1500x64xdlf16>, !common.dynamic_index<i16>) -> () loc(callsite("-":33:3 at fused<array<i64: 340>>["rbln_cache_update":0:0]))
[2026-05-14 16:27:12] [2160342][2160342] [E] RTOSAInferShape pass failed
[2026-05-14 16:27:12] [2160342][2160342] [E] mlir::detail::OpToOpPassAdaptor pass failed
[2026-05-14 16:27:12] [2160342][2160342] [E] mlir::detail::OpToOpPassAdaptor pass failed
```

File: src/optimum/rbln/transformers/models/whisper/modeling_whisper.py [hunk 2]
```diff
@@ -290,6 +290,7 @@ def _update_rbln_config(
             rbln_config.dec_max_seq_len = model_config.max_length
 
         cls._update_paged_attention_config(model_config, rbln_config)
+        breakpoint()
```
Comment (rebel-eunji) [inline src/optimum/rbln/transformers/models/whisper/modeling_whisper.py]:
```suggestion
```

**Comment by rebel-eunji** [comment]
@rebel-kblee Please review this PR.

**Comment by rebel-kblee** [comment]
이게 제가 이해한게 맞을까요...?
- 이렇게 하면 항상 dummy block 1개가 추가 할당 되는거 아닌가요...? (generate 사용자도..?) (설명 필요...?)
- dummy block은 항상 마지막이었죠..?
- configuration_whisper.py에 kvcache_num_blockss의 default값이 b+1이 되어야 할 것 같습니다
@rebel-eunji @rebel-thkim

**Review by rebel-kblee — APPROVED** [review]
lgtm 👍
